pyaerocom/io/read_eea_aqerep.py

```diff
@@ -0,0 +1,538 @@
+
+        Parameters
+        ----------
+        pattern : str, optional
+            file name pattern applied to search
+
+        Returns
+        -------
+        list
+            list containing retrieved file locations
+
+        Raises
+        ------
+        IOError
+            if no files can be found
+        """
+        import glob, os
+        from pyaerocom._lowlevel_helpers import list_to_shortstr
+        from pyaerocom.exceptions import DataSourceError
+
+        if pattern is None:
+            const.print_log.warning('_FILEMASK attr. must not be None...'
+                                    'using default pattern *.* for file search')
+            pattern = '*.*'
+        self.logger.info('Fetching data files. This might take a while...')
+        files = sorted(glob.glob(os.path.join(self.DATASET_PATH,
+                                              pattern), recursive=True))
+        if not len(files) > 0:
+            all_str = list_to_shortstr(os.listdir(self.DATASET_PATH))
+            raise DataSourceError('No files could be detected matching file '
+                                  'mask {} in dataset {}, files in folder {}:\n'
+                                  'Files in folder:{}'.format(pattern,
+                                  self.dataset_to_read,
+                                  self.DATASET_PATH,
+                                  all_str))
+        self.files = files
+        return files
+
+    def get_station_coords(self, meta_key):
+        """
+        get a station's coordinates
+
+        Parameters
+        ----------
+        meta_key : `str`
+            string with the internal station key
+        """
+
+        ret_data = {}
+        ret_data['latitude'] = self._metadata[meta_key][self.LATITUDENAME]
+        ret_data['longitude'] = self._metadata[meta_key][self.LONGITUDENAME]
+        ret_data['altitude'] = self._metadata[meta_key][self.ALTITUDENAME]
+        return ret_data
+
+    def read(self, vars_to_retrieve=None,
+             files=None,
+             first_file=None,
+             last_file=None,
+             metadatafile=None):
+        """Method that reads list of files as instance of :class:`UngriddedData`
+
+        Parameters
+        ----------
+        vars_to_retrieve : :obj:`list` or similar, optional
+            List containing variable IDs that are supposed to be read. If None,
+            all variables in :attr:`PROVIDES_VARIABLES` are loaded.
+        files : :obj:`list`, optional
+            List of files to be read. If None, then the file list used is the
+            returned from :func:`get_file_list`.
+        first_file : :obj:`int`, optional
+            Index of the first file in :obj:'file' to be read. If None, the
+            very first file in the list is used.
+        last_file : :obj:`int`, optional
+            Index of the last file in :obj:'file' to be read. If None, the very
+            last file in the list is used.
```

Comment:
Maybe add also description for parameter `metadatafile`

Reply:
done